### PR TITLE
DCP 632: Fix link map when there is file to file linking

### PR DESCRIPTION
**Pull request by aaclan-ebi**
dcp-632
ebi-ait/dcp-ingest-central#632

### CODE REVIEW

**Review by jacobwindsor — APPROVED** [review]

File: src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java [hunk 1]
```diff
@@ -52,27 +52,39 @@ public void testSubmissionLinkMap() {
...
         donor.addAsInputToProcess(process1);
         cellSuspension.addAsInputToProcess(process2);
+        sequencingFile.addAsInputToProcess(process3);
 
         process1.addProtocol(collectionProtocol);
         process2.addProtocol(sequencingProtocol);
+        process3.addProtocol(analysisProtocol);
 
-        when(processRepository.findBySubmissionEnvelope(submissionEnvelope)).thenReturn(Stream.of(process1, process2));
+        when(processRepository.findBySubmissionEnvelope(submissionEnvelope)).thenReturn(Stream.of(process1, process2, process3));
```
Comment (amnonkhen) [inline src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java:77]: Why does this test need to be a `SpringBootTest`?
Could it be only a `DataMongoTest`?
This way you would just save everything in the repositories, and there would not be a need to use the `when().thenReturn()`

File: src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java [hunk 2]
```diff
@@ -81,10 +93,15 @@ public void testSubmissionLinkMap() {
         assertThat(submissionLinkMap).isNotNull();
         assertThat(submissionLinkMap.processes.get("donor-specimen-process").protocols).isEqualTo(new HashSet<>(Arrays.asList("collectionProtocol")));
         assertThat(submissionLinkMap.processes.get("donor-specimen-process").inputBiomaterials).isEqualTo(new HashSet<>(Arrays.asList("donor")));
-        assertThat(submissionLinkMap.processes.get("cellSuspension-file-process").protocols).isEqualTo(new HashSet<>(Arrays.asList("sequencingProtocol")));
-        assertThat(submissionLinkMap.processes.get("cellSuspension-file-process").inputBiomaterials).isEqualTo(new HashSet<>(Arrays.asList("cellSuspension")));
+        assertThat(submissionLinkMap.processes.get("cellSuspension-sequenceFile-process").protocols).isEqualTo(new HashSet<>(Arrays.asList("sequencingProtocol")));
+        assertThat(submissionLinkMap.processes.get("cellSuspension-sequenceFile-process").inputBiomaterials).isEqualTo(new HashSet<>(Arrays.asList("cellSuspension")));
+        assertThat(submissionLinkMap.processes.get("cellSuspension-sequenceFile-process").inputFiles).isEmpty();
```
Comment (amnonkhen) [inline src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java:98]: Have you considered less duplication by using `extracting()` or `from()`?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java (hunk 1 above)
Comment (aaclan-ebi) [reply src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java]: I've converted them as Spring's integration test and just autowired the repository objects as I'm using them in the SubmissionLinkMapController.  It did avoid mocking the repository methods in the tests.

File: src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java (hunk 2 above)
Comment (aaclan-ebi) [reply src/test/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapControllerTest.java]: I believe they're not applicable here.

**Review by jacobwindsor — APPROVED** [review]

**Review by yusra-haider — APPROVED** [review]

**Pull request merged**